### ISSUE TITLE
Cannot assign empty tuple to `Tuple[_T, ...]`

### ISSUE THREAD

**Issue by 0xjc** [microsoft/pyright#2821]
```
# pyright: strict
from typing import Generic, Tuple, TypeVar

_T = TypeVar("_T")


class X(Generic[_T]):
    def __init__(self):
        self._x: Tuple[_T, ...] = ()
```

fails with:

```
error: Cannot assign member "_x" for type "X[_T@X]"
    Expression of type "tuple[()]" cannot be assigned to member "_x" of class "X[_T@X]"
      Tuple entry 1 is incorrect type
        Type "Never" cannot be assigned to type "_T@X" (reportGeneralTypeIssues)
```

Appears to be a new error in pyright 1.1.204.

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.205, which I just published. It will also be included in the next release of pylance.
